assets/stylesheets/_components.scss

```diff
@@ -7,504 +7,504 @@
-@import 'my-sites/stats/stats-chart-tabs/style';
-@import 'my-sites/stats/stats-comments/style';
-@import 'my-sites/stats/stats-countries/style';
-@import 'my-sites/stats/stats-date-picker/style';
-@import 'my-sites/stats/stats-error/style';
-@import 'my-sites/stats/stats-module/style';
-@import 'my-sites/stats/stats-page-placeholder/style';
-@import 'my-sites/stats/stats-period-navigation/style';
-@import 'my-sites/stats/stats-post-likes/style';
-@import 'my-sites/stats/stats-post-summary/style';
-@import 'my-sites/stats/stats-site-overview/style';
-@import 'my-sites/stats/stats-tabs/style';
-@import 'my-sites/stats/stats-views/style';
-@import 'my-sites/theme/style';
-@import 'my-sites/theme/theme-download-card/style';
-@import 'my-sites/themes/current-theme/style';
-@import 'my-sites/themes/style';
-@import 'my-sites/themes/themes-magic-search-card/style';
-@import 'my-sites/themes/theme-upload/style';
-@import 'my-sites/themes/themes-selection-header/style';
-@import 'my-sites/themes/themes-banner/style';
-@import 'my-sites/upgrade-nudge/style';
-@import 'my-sites/checkout/cart/style';
-@import 'my-sites/checkout/checkout/stored-card';
-@import 'my-sites/checkout/checkout/style';
-@import 'my-sites/checkout/checkout/subscription-text';
-@import 'my-sites/checkout/checkout-thank-you/style';
-@import 'my-sites/checkout/checkout-thank-you/google-voucher/style';
-@import 'my-sites/checkout/gsuite-nudge/style';
-@import 'my-sites/domains/components/domain-warnings/style';
-@import 'my-sites/domains/domain-management/components/designated-agent-notice/style';
-@import 'my-sites/domains/domain-management/transfer/transfer-out/style';
-@import 'my-sites/domains/domain-management/transfer/transfer-to-other-user/style';
-@import 'my-sites/domains/domain-search/style';
-@import 'my-sites/resume-editing/style';
-@import 'my-sites/stats/stats-list/style';
-@import 'notices/style';
-@import 'my-sites/domains/domain-management/style';
-@import 'my-sites/domains/domain-management/email/style';
-@import 'my-sites/domains/domain-management/components/email-verification/style';
-@import 'my-sites/domains/domain-management/components/inbound-transfer-verification/style';
-@import 'my-sites/domains/domain-management/components/icann-verification/style';
-@import 'notifications/style';
-@import 'reader/conversations/style';
-@import 'reader/discover/style';
-@import 'reader/following/style';
-@import 'reader/following-manage/style';
-@import 'reader/stream/style';
-@import 'reader/like-button/style';
-@import 'reader/list-gap/style';
-@import 'reader/list-stream/style';
-@import 'reader/post-excerpt-link/style';
-@import 'reader/reading-time/style';
-@import 'reader/search-stream/style';
-@import 'reader/sidebar/style';
-@import 'reader/site-stream/style';
-@import 'reader/tag-stream/style';
-@import 'reader/style';
-@import 'reader/update-notice/style';
+@import '../../client/account-recovery/components/account-recovery-error-message/style.scss';
```

Comment:
I don't think we need to specifically add the file format `.scss` for importing any of the `style.scss` files. We would only need to do it for files that aren't named `style.scss`, like `components/forms/sortable-list/index.scss` for example. An alternative would be to rename all those special case files (we have like 10 of them) to `style.scss` where possible.

Reply:
we had a discussion about this in an earlier exploration and I think there's much value in providing the explicit extension to assets. do you see a mistake we're making?

https://github.com/Automattic/wp-calypso/pull/26325#issuecomment-408838869

Reply:
I don't have hard feelings about this, but I'd love it if we could keep it shorter. I'm willing to learn more about the value this brings - could you please provide some details?

Reply:
the value is explicit path resolution. we have invited bugs in two primary forms…

 - modules with the same resolution path exist in both `client` and `server` and it's not clear until runtime (unless you know the code inside and out) which one will be resolved under what circumstances
 - modules share the same base name but differ in extension and then you have to understand our webpack config and build process in order to understand the source code

the value is that there's less implicit activity going on which means that the source code maps more directly to something you can see in the repository